group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -874,4 +1021,34 @@ public void replay(
             consumerGroup.updateMember(newMember);
         }
     }
+
+    /**
+     * A new metadata image is available.
+     *
+     * @param newImage  The new metadata image.
+     * @param delta     The delta image.
+     */
+    public void onNewMetadataImage(MetadataImage newImage, MetadataDelta delta) {
+        metadataImage = newImage;
+
+        // Notify all the groups subscribed to the created, updated or
+        // deleted topics.
+        Set<String> allGroupIds = new HashSet<>();
+        delta.topicsDelta().changedTopics().forEach((topicId, topicDelta) -> {
+            String topicName = topicDelta.name();
+            Set<String> groupIds = groupsByTopics.get(topicName);
+            if (groupIds != null) allGroupIds.addAll(groupIds);
+        });
+        delta.topicsDelta().deletedTopicIds().forEach(topicId -> {
+            TopicImage topicImage = delta.image().topics().getTopic(topicId);
+            Set<String> groupIds = groupsByTopics.get(topicImage.name());
+            if (groupIds != null) allGroupIds.addAll(groupIds);
```

Comment:
we don't want to add all the deleted topics do we?

Reply:
Oh i misunderstood -- this is to add to the list of groups to notify. 👍

Reply:
Correct. I also simplified this code.